tools/build-for-arm.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+#
+# Copyright 2021 Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -ex
+
+CURR_DIR=$(dirname "${BASH_SOURCE[0]}")
+
+if [ ! -d ${CURR_DIR}/istio ]; then
+    git clone https://github.com/istio/istio.git &&
+    git clone https://github.com/istio/proxy.git &&
+    git clone https://github.com/istio/tools.git
+fi
+
+# Build build-tools and build-tools-proxy
+pushd ${CURR_DIR}/tools/docker/build-tools
+if [ -z "$(docker images |grep "build-tools-proxy" | grep "master-latest")" ]; then
+    DRY_RUN=1 time ./build-and-push.sh
+fi
+popd
+
+# Build Istio binary
+pushd ${CURR_DIR}/istio
+TARGET_ARCH=arm64 IMAGE_VERSION=master-latest make build
+popd
+
+# Build Envoy and install envoy
+pushd ${CURR_DIR}/proxy
+export PROXY_REPO_SHA=$(git rev-parse HEAD)
+popd
+if [ ! -f ${CURR_DIR}/istio/out/linux_arm64/release/envoy ]; then
+    docker rm -f istio-proxy-build || true
+    docker run --name istio-proxy-build -it -v $(realpath ${CURR_DIR}/proxy):/work -w /work -u $(id -u) gcr.io/istio-testing/build-tools-proxy:master-latest make build
+    docker cp istio-proxy-build:/work/bazel-bin/src/envoy/envoy ${CURR_DIR}/istio/out/linux_arm64/release/envoy-${PROXY_REPO_SHA}
```

Comment:
I think `exportcache` target is supposed to do this. Not sure that works as expected though

Reply:
yes, exportcache would copy envoy binary to its out/linux_arm64, and we still need to copy it to istio/out/linux_arm64.

Reply:
Updated.